src/Types/ListType.php

```diff
@@ -23,9 +25,9 @@ public function areKeysInvalid(): bool
 	/**
 	 * @param string|int $key
 	 */
-	public function addInvalidItem($key, Type $itemType): void
+	public function addInvalidItem($key, WithTypeAndValue $typeAndValue): void
 	{
-		$this->invalidItems[$key] = $itemType;
+		$this->invalidItems[$key] = $typeAndValue->getInvalidType();
```

Comment:
`WithTypeAndValue` should be stored in invalidItems instead of getting `Type` here